### PR TITLE
Fix logging error that includes 'No error' message when throwing exception.

### PR DESCRIPTION
**Pull request by kamieater1240**
The parameter 'code' (which is a curl function) in the CHECKCURLCODE macro will be evaluated twice (which calls the curl function twice) if it is used in the if condition check and passed in the curl_easy_strerror.
Therefore, we use returnCode to store the CURLcode returned by the parameter 'code', and pass the returnCode to the curl_easy_strerror function to prevent it being executed twice.

### CODE REVIEW

File: CHANELOG.md [hunk 1]
```diff
@@ -1,5 +1,13 @@
 # Changelog
 
+## 0.73.5 (2024-08-12)
```
Comment (woswos) [inline CHANELOG.md:3]: @kamieater1240 Let's merge these changelogs into a single version update.

Comment (kamieater1240) [reply CHANELOG.md]: OK, changed back to the single version update.

Comment (woswos) [reply CHANELOG.md]: Thank you!

**Review by woswos — APPROVED** [review]
Pipeline ID: 908417

File: src/common.h [hunk 2]
```diff
@@ -187,16 +187,27 @@ inline static unsigned long long GetNanoPerformanceTime()
 #endif
 
 #define GETCONTROLLERIMPL() ControllerClientImplPtr controller = boost::dynamic_pointer_cast<ControllerClientImpl>(GetController());
-#define CHECKCURLCODE(code, msg) if (code != CURLE_OK) { \
-        throw MujinException(boost::str(boost::format("[%s:%d] curl function %s with error '%s': %s")%(__PRETTY_FUNCTION__)%(__LINE__)%(msg)%curl_easy_strerror(code)%_errormessage), MEC_HTTPClient); \
+#define CHECKCURLCODE(code, msg) { \
```
Comment (ziyan) [inline src/common.h:190]: Please put the comment outside of the define.

Please use `do { ... } while (false)` to scope instead of just `{ ... }`.

Comment (kamieater1240) [reply src/common.h]: @ziyan I edited the scope to the do { ... } while (false) form. Can you please check when you are free? Thanks.

**Comment by ziyan** [comment]
@kamieater1240 Thank you!

**Pull request merged**